File: Linked_List/DeleteMiddleNode.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* deleteMiddle(struct ListNode* head) {
    if(!head || (!head->next))  return NULL; // one element or empty list

    struct ListNode *mid = head, *upper = head, *upperPrev = NULL;

    while (upper && upper->next)
    {
        upperPrev = mid;
        mid = mid->next;
        upper = upper->next->next;
    }

    upperPrev->next = mid->next;
    free(mid);

    return head;
}

File: Linked_List/DeleteMiddleNode.java
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* deleteMiddle(struct ListNode* head) {
    int count=0,i;
    struct ListNode *temp=head, *temp2=head, *temp3=head;
    while(temp!=NULL)
    {
        count++;
        temp=temp->next;
    }
    printf(" total nodes is: %d\n",count);
    int a;
    a=count/2;
    printf("middle index number of nodes is: %d",a);
    for(i=0;i<a;i++)
    {
        temp2=temp3;
        temp3=temp3->next;
    }
    temp2->next=temp3->next;
    free(temp3);
    return head;
}
